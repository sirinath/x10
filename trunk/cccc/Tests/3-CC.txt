# CC example from Doug Lea's and Vijay Saraswat's writeups

1 1 w x 1
2 2 r x 1
2 3 w y 1
3 4 r y 1
3 5 r x 0

1.1 -> 2.2
2.3 -> 3.4
3.5 -> 1.1
