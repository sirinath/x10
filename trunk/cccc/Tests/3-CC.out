1.1 w x 1
2.2 r x 1
2.3 w y 1
3.4 r y 1
3.5 r x 0
# Manually specified dependencies
1.1 -> 2.2
2.3 -> 3.4
3.5 -> 1.1
# Program-order dependencies
2.2 -> 2.3
3.4 -> 3.5
# Propagated dependencies
1.1 -> 2.3
2.2 -> 3.4
2.2 -> 3.5
2.3 -> 3.5
3.4 -> 1.1
Happens-before dependencies for task 1
1.1 -> 2.2
1.1 -> 2.3
2.2 -> 2.3
3.4 -> 1.1
3.4 -> 3.5
3.5 -> 1.1
Happens-before dependencies for task 2
1.1 -> 2.2
1.1 -> 2.3
2.2 -> 2.3
2.2 -> 3.4
2.2 -> 3.5
2.3 -> 3.4
2.3 -> 3.5
3.4 -> 3.5
Happens-before dependencies for task 3
2.2 -> 2.3
2.2 -> 3.4
2.2 -> 3.5
2.3 -> 3.4
2.3 -> 3.5
3.4 -> 1.1
3.4 -> 3.5
3.5 -> 1.1
No cycles found: plausible CCCC execution
